internal/exec/path_utils.go

```diff
@@ -80,8 +85,189 @@ func constructHelmfileComponentVarfileName(info schema.ConfigAndStacksInfo) stri
+					folderNames = append(folderNames, filepath.Join(dir.Name(), subDir.Name()))
+				}
+			}
+		}
+	}
+
+	return folderNames, nil
+}
+
+// DeleteFilesAndFoldersRecursive deletes specified files and folders from the base path,
+// including those found in immediate subdirectories.
+func deleteFilesAndFoldersRecursive(basePath string, items []string) error {
+	// First, delete files and folders directly under the base path
+	for _, item := range items {
+		fullPath := filepath.Join(basePath, item)
+
+		// Attempt to delete the file or folder
+		err := os.RemoveAll(fullPath)
+		if err != nil {
+			u.LogWarning(schema.CliConfiguration{}, fmt.Sprintf("Error deleting %s: %v", fullPath, err))
+			continue
+		}
+		u.LogInfo(schema.CliConfiguration{}, fmt.Sprintf("Deleted %s", item))
+
+	}
+
+	// Now, delete matching files and folders from immediate subdirectories
+	entries, err := os.ReadDir(basePath)
+	if err != nil {
+		return fmt.Errorf("error reading the base path %s: %v", basePath, err)
+	}
+
+	for _, entry := range entries {
+		// Only proceed if the entry is a directory
+		if entry.IsDir() {
+			subDirPath := filepath.Join(basePath, entry.Name())
+
+			for _, item := range items {
+				fullPath := filepath.Join(subDirPath, item)
+				// Attempt to delete the file or folder
+				err := os.RemoveAll(fullPath)
+				if err != nil {
+					u.LogWarning(schema.CliConfiguration{}, fmt.Sprintf("Error deleting %s: %v", item, err))
+					continue
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// Helper functions to improve readability and maintainability
+func cleanAllComponents(cliConfig schema.CliConfiguration, filesToClear []string, componentPath string, force bool) error {
+	u.LogWarning(cliConfig, fmt.Sprintf("Deleting all local Terraform state files for all components in path %s", componentPath))
+	if !force {
+		message := "This will delete all local Terraform state files for all components.\nAre you sure you want to proceed?"
+		confirm, err := confirmDeleteTerrFormLocal(message)
+		if err != nil {
+			return err
+		}
+		if !confirm {
+			u.LogWarning(cliConfig, "Operation canceled.")
+			return nil
+		}
+	}
+	// get relative path form base path
+
+	err := deleteFilesAndFoldersRecursive(componentPath, filesToClear)
+	if err != nil {
+		u.LogWarning(cliConfig, err.Error())
+	}
+
+	return nil
+}
+
+func cleanSpecificComponent(cliConfig schema.CliConfiguration, componentPath string, filesToClear []string, force bool, componentName string, baseComponentName string) error {
+	u.LogWarning(cliConfig, fmt.Sprintf("Deleting local Terraform state files for component in path %s", componentPath))
+
+	if !force {
+		if !force {
+			message := fmt.Sprintf("This will delete all local Terraform state files for component %s base %s.\nAre you sure you want to proceed?", componentName, baseComponentName)
+			confirm, err := confirmDeleteTerrFormLocal(message)
+			if err != nil {
+				return err
+			}
+			if !confirm {
```

Comment:
Why do we have this repeated so many times

Reply:
to be confirm message dynamic but I will handle it